lib/iris/tests/unit/common/lenient/test__Lenient.py

```diff
@@ -88,54 +91,55 @@ def service2():
                 _qualname(client2): (_qualname(service1), _qualname(service2)),
             }
         )
-        self.assertEqual(self.expected, lenient.__dict__)
+        assert lenient.__dict__ == self.expected
 
 
-class Test___call__(tests.IrisTest):
-    def setUp(self):
+class Test___call__:
+    @pytest.fixture(autouse=True)
+    def _setup(self):
         self.client = "myclient"
         self.lenient = _Lenient()
 
     def test_missing_service_str(self):
-        self.assertFalse(self.lenient("myservice"))
+        assert bool(self.lenient("myservice")) is False
```

Comment:
Possibly a bit of a rant going here, but here's my thinking ..

I believe that "truthiness" in python is just always whatever "bool(x)" returns, and what the "if" and "assert" statements consider true or false is _precisely_ "whatever bool says".  
So you shouldn't write "if bool(x)" or "assert bool(x)" because that is already a guaranteed part of what "if" and "assert" are defined to do.  
And if "if bool(x)" is redundant, because that's exactly what "if" does already, then also "if bool(x) is True" is _doubly_ redundant, because that's exactly what "bool" does.
As far as I understand it, you can never have an x which passes "x ^ bool(x)",  or "bool(x) & bool(x) is not True" or "not bool(x) & bool(x) is not False".  Because "bool" is the _only_ decider of "truthiness" in Python, and it only ever returns True or False.

And I think that whatever goes for "if" here goes for "assert" just as well.
I looked up "assert" in the Python docs, and they do state that it _is_ [equivalent to "if not x: raise"](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement)